core/dispatch_json.js

```diff
@@ -0,0 +1,103 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+
+((window) => {
+  // Available on start due to bindings.
+  const core = window.Deno.core;
+
+  let errorCallback;
+
+  function assert(cond) {
+    if (!cond) {
+      throw Error("assert");
+    }
+  }
+
+  function createResolvable() {
+    let resolve;
+    let reject;
+    const promise = new Promise((res, rej) => {
+      resolve = res;
+      reject = rej;
+    });
+    promise.resolve = resolve;
+    promise.reject = reject;
+    return promise;
+  }
+
+  // Using an object without a prototype because `Map` was causing GC problems.
+  const promiseTable = Object.create(null);
+  let _nextPromiseId = 1;
+
+  function nextPromiseId() {
+    return _nextPromiseId++;
+  }
+
+  function decode(ui8) {
+    const s = core.decode(ui8);
+    return JSON.parse(s);
+  }
+
+  function encode(args) {
+    const s = JSON.stringify(args);
+    return core.encode(s);
+  }
+
+  function setErrorCb(errorCb) {
```

Comment:
This is going to be a problem if we want to use this with plugins as well. Plugins are not and should not be aware of our error kinds defined in cli, and even if they where if we change them from one version to another it will break plugins.